ladybug/resultmesh.py

```diff
@@ -0,0 +1,223 @@
+# coding=utf-8
+from __future__ import division
+
+from .legend import Legend, LegendParameters
+
+from .datatype.base import DataTypeBase
+
+from ladybug_geometry.geometry3d.pointvector import Point3D
+from ladybug_geometry.geometry3d.plane import Plane
+from ladybug_geometry.geometry3d.mesh import Mesh3D
+
+
+class ResultMesh(object):
+    """Ladybug colored result mesh with legend.
+
+    Properties:
+        values
+        colored_mesh
+        legend_parameters
+        legend
+        value_colors
+        lower_title_location
+        upper_title_location
+        data_type
+        unit
+    """
+
+    def __init__(self, values, mesh, legend_parameters=None, data_type=None, unit=None):
+        """Initialize result mesh.
+
+        Initializing a legend and colored mesh this way will generate default
+        legend base points, text points, and general scaling that makes sense
+        given the input mesh.
+
+        Args:
+            values: A List or Tuple of numerical values that will be used to
+                generate the legend and colors.
+            mesh: A Mesh3D object, with a number of faces or vertices that
+                match the number of input values and will be colored with
+                restults.
+            legend_parameters: An Optional LegendParameter object to override
+                default parameters of the legend.
+        """
+        # check the inputs
+        assert isinstance(mesh, Mesh3D), \
```

Comment:
This can also be useful for other visualizations like sun-path with a legend. Maybe we should ask for a bounding box more than a Mesh input.

Also I feel this class can be a method to give suggestive values. Right now the logic is built-in and so making changes to it is hard. I can see it as a function that takes a geometry and legend parameters and returns the legend with the suggested location which user can modify before pushing it for visualization.

Reply:
I have been going back and forth about whether this object should be specific to displaying colored meshes of studies (ie. daylight, thermal comfort, color zones) or whether it should be generalized for all geometry so we can use it for sunpath, wind rose, hourly chart, etc.  You are right that all that we would have to do for the latter is to have people input a min and max point instead of a mesh (I think just min_point and max_point is better than a special bounding_box object because specifying the points allows more flexibility).  And I guess the only extra code people would need to add outside of the general object to get it to work for colored meshes is pass the `mesh.min`, `mesh.max`, and write `mesh.colors = legend.value_colors`.  Ok. I will re-make it so that it's a bit more general.